### ISSUE TITLE
EntityEx, moving more info to extended netobjects

### ISSUE THREAD

**Issue by trml** [ddnet/ddnet#5860]
Since netobjects are easier to extend now, and since EntityEx still has its issues/limitations, it can perhaps be considered to move the info from entityex (i.e. switch number) to DDNetLaser, DDNetProjectile, and so on.

Some thoughts/proposals:
- Adding the switch number to DDNetLaser. It would also be nice to distinguish both between all the laser types (this is already added in ddnetlaser, but only for the door type), and to add a "SubType" field to further distinguish for example types of draggers.
  - Another possibility would be to have separate netobjects for each type and keep ddnetlaser for lasers, but a type+sybtype scheme would be flexible enough to support the current variants, and also cleaner than the scheme I originally added to EntityEx (a single type enum with more detailed types)

- For DDNetProjectile I'm not sure if it is straightforward to add new fields, due to the compatibility with old projectile sizes? One alternative could be to add a completely new netobject (benefit: an opportunity to fix the problems with the old netobject, cons: would still have to support the old one)

- Add a DDNetPickup with a switch number field

Downsides: The client would still have to support EntityEx for some time to correctly display doors and pickups/projectiles in switch layer on servers that send EntityEx.

Dropping support for EntityEx from the server (by reverting to the old way of showing pickups/doors for older clients) would only affect prediction of switched pickups/projectiles and would be less noticeable.

**Comment by fokkonaut** [comment]
> Downsides: The client would still have to support EntityEx for some time to correctly display doors and pickups/projectiles in switch layer on servers that send EntityEx.
We should drop that. Its not "too" many versions, and they wouldnt miss a lot compared to the clients before.

**Comment by trml** [comment]
@Learath2, @heinrich5991: Thoughts on adding a new projectile netobject and on reusing ddnetlaser for other laser types (vs adding new netobjects for other laser entities later)?
I could open a pr if the general idea looks ok.

**Comment by fokkonaut** [comment]
> @Learath2, @heinrich5991: Thoughts on adding a new projectile netobject and on reusing ddnetlaser for other laser types (vs adding new netobjects for other laser entities later)? I could open a pr if the general idea looks ok.

I implemented DDNetLaser to be used by other lasers. Thats the case already

**Comment by trml** [comment]
@fokkonaut That's true. Also, there is perhaps no disadvantage to use the ddnetlaser for communicating more info about all current laser types, since any future type that don't fit it would require a new netobject (or an extension) either way.

**Comment by AssassinTee** [comment]
This is closed by #6560 why is this not closed since 2 years?
